src/app/utilities/common.ts

```diff
@@ -42,14 +43,28 @@ export const throttle = (
   };
 };
 
-export const createMetadata = ({
-  title,
-  keywords,
-  description,
-  canonical,
-  type,
-  images,
-}: MetadataParams) => {
+export const createMetadata = ({ key, params }: MetadataKeyValuePair) => {
+  let { title, description, keywords, canonical, type, images } = metadata[key];
+
+  //extend with metadataParams
+  title += title ? ` | ${params?.title || ''}` : params?.title || '';
+
+  description += description
+    ? ` | ${params?.description || ''}`
+    : params?.description || '';
+
+  keywords += keywords
+    ? ` | ${params?.keywords || ''}`
+    : params?.keywords || '';
+
+  canonical += canonical
+    ? ` | ${params?.canonical || ''}`
+    : params?.canonical || '';
```

Comment:
Create a base config based on the `metadata[key]`. Then do a shallow merge. Like `config = { ...baseConfig, ...extension }` - passing the second argument will denote that we want to override the built-in values.